rasa/nlu/test.py

```diff
@@ -1319,94 +1308,34 @@ def get_eval_data(
-
-    Args:
-        interpreter: the interpreter
-
-    Returns: entity extractor names
-    """
-    from rasa.nlu.extractors.extractor import EntityExtractor
-    from rasa.nlu.classifiers.diet_classifier import DIETClassifier
-
-    extractors = set()
-    for c in interpreter.pipeline:
-        if isinstance(c, EntityExtractor):
-            if isinstance(c, DIETClassifier):
-                if c.component_config[ENTITY_RECOGNITION]:
-                    extractors.add(c.name)
-            else:
-                extractors.add(c.name)
-
-    return extractors - ENTITY_PROCESSORS
-
-
-def is_entity_extractor_present(interpreter: Interpreter) -> bool:
-    """Checks whether entity extractor is present."""
-    extractors = get_entity_extractors(interpreter)
-    return len(extractors) > 0
-
-
-def is_intent_classifier_present(interpreter: Interpreter) -> bool:
-    """Checks whether intent classifier is present."""
-    from rasa.nlu.classifiers.classifier import IntentClassifier
-
-    intent_classifiers = [
-        c.name for c in interpreter.pipeline if isinstance(c, IntentClassifier)
-    ]
-    return intent_classifiers != []
-
-
-def is_response_selector_present(interpreter: Interpreter) -> bool:
-    """Checks whether response selector is present."""
-
-    from rasa.nlu.selectors.response_selector import ResponseSelector
-
-    response_selectors = [
-        c.name for c in interpreter.pipeline if isinstance(c, ResponseSelector)
-    ]
-    return response_selectors != []
-
-
-def get_available_response_selector_types(
-    interpreter: Interpreter,
-) -> List[Optional[Text]]:
-    """Gets all available response selector types."""
-    from rasa.nlu.selectors.response_selector import ResponseSelector
-
-    response_selector_types: List[Optional[Text]] = [
-        c.retrieval_intent
-        for c in interpreter.pipeline
-        if isinstance(c, ResponseSelector)
-    ]
-
-    return response_selector_types
-
-
-def remove_pretrained_extractors(pipeline: List[Component]) -> List[Component]:
-    """Remove pre-trained extractors from the pipeline.
-
-    Remove pre-trained extractors so that entities from pre-trained extractors
-    are not predicted upon parsing.
+def get_active_entity_extractors(
+    entity_results: List[EntityEvaluationResult],
+) -> Set[Text]:
+    """Finds the names of entity extractors from the EntityEvaluationResults."""
+    extractors: Set[Text] = set()
+    for result in entity_results:
+        for prediction in result.entity_predictions:
+            if EXTRACTOR in prediction:
+                extractors.add(prediction[EXTRACTOR])
+    return extractors
 
-    Args:
-        pipeline: the pipeline
 
-    Returns:
-        Updated pipeline
-    """
-    pipeline = [c for c in pipeline if c.name not in PRETRAINED_EXTRACTORS]
-    return pipeline
+def remove_entities_of_extractors(
+    result_dict: Dict[Text, Any], extractor_names: Set[Text]
```

Comment:
optional: Seems a little bit less vague that `result_dict` to me
```suggestion
    nlu_parse_result: Dict[Text, Any], extractor_names: Set[Text]
```